File: Pascal/6kyu/Sum_of_Digits_/_Digital_Root.pas
unit Kata;

interface

function DigitalRoot (n: Integer): Integer;

implementation

function DigitalRoot (n: Integer): Integer;
var m, Res: Integer;
begin
   Res := 0;
  while n > 0 do
  begin 
     m := n mod 10;
     Res:=Res + m;
     n := Trunc((n-m)/10);
  end;
  if Res > 9 then Res:= DigitalRoot(Res);
  
  Result:=Res;

end;

end.

File: Java/6kyu/Sum of Digits - Digital Root/_Digital_Root.java
public class DRoot {
    public static int digital_root(int n) {
      int res = 0;
      while(n > 0){
        res += (n%10);
        n = n/10;
      }
      if(res > 9) res = digital_root(res);
      return res;
    }
  }